deeplake/core/dataset/dataset.py

```diff
@@ -1137,6 +1138,8 @@ def commit(self, message: Optional[str] = None, allow_empty=False) -> str:
         return self._commit(message)
 
     @deeplake_reporter.record_call
+    @invalid_view_op
+    @suppress_iteration_warning
```

Comment:
can we also add the warning suppression to ds.commit?

Reply:
I don't see how commit can lead to iteration warning